torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -153,11 +153,18 @@ void initJitBackendBindings(PyObject* module) {
         /*shouldMangle=*/true);
 
     // Generate attributes.
-    // This is the original cloned and preprocessed module.
+    // This is the preprocessed module.
+    // For backwards compatibility, for backends that implement preprocessing in
+    // the backend interface rather than as a separate function, we just pass
+    // the cloned original Module.
     loweredModule.register_attribute(
         "__processed_module",
         AnyType::get(),
-        cloned_module._ivalue(),
+        detail::hasBackendPreprocessFunction(backend_name)
+            ? detail::getBackendPreprocessFunction(backend_name)(
```

Comment:
Aha, the compilation can happen here.

Reply:
That's right.
Here we set the value by calling the BackendPreprocessFunction (if available).
Otherwise we do as before and first set the cloned original module, and then below we add the __preprocess function and call it (as before).

Reply:
Discussed how this works offline.
Resolving the comment.